src/classes/BDI_DataImportService.cls

```diff
@@ -1725,66 +1725,81 @@ global with sharing class BDI_DataImportService {
     */
     private void importCampaigns() {
 
-        // since we don't currently fill in a campaign lookup or status on the DI record,
-        // there is no work to do in Dry Run.
-        if (isDryRun) {
-            return;
-        }
-
         // need to find all the Campaigns that we might need
-        Set<String> setStrCampaign = new Set<String>();
+        Set<String> uniqueCampaignNames = new Set<String>();
+        Set<Id> setCampaignId = new Set<Id>();
+
         for (DataImport__c dataImport : listDI) {
             // skip di's that already have an error
             if (dataImport.Status__c == BDI_DataImport_API.bdiFailed) {
                 continue;
             }
 
             if (dataImport.Donation_Campaign_Name__c != null) {
-                setStrCampaign.add(dataImport.Donation_Campaign_Name__c);
+                uniqueCampaignNames.add(dataImport.Donation_Campaign_Name__c);
             }
 
         }
 
         // add the existing campaigns to our map
-        for (Campaign cmp : [SELECT Id, Name FROM Campaign WHERE Name IN :setStrCampaign]) {
-            mapStrNameToCampaignId.put(cmp.Name, cmp.Id);
+        for (Campaign cmp : [SELECT Id, Name FROM Campaign WHERE Name IN :uniqueCampaignNames]) {
+            campaignNameToCampaignId.put(cmp.Name, cmp.Id);
         }
 
         // create any new campaigns
-        List<Campaign> listCmpNew = new List<Campaign>();
-        List<DataImport__c> listDINew = new List<DataImport__c>();
-        setStrCampaign.clear();
+        List<Campaign> campaignsToInsert = new List<Campaign>();
+        List<DataImport__c> dataImportsWithNewCampaigns = new List<DataImport__c>();
+        uniqueCampaignNames.clear();
```

Comment:
Do we need to clear this out?

Reply:
Yes, this is part of the logic the code relies on below. As we loop through each match we add them back to this list, and it prevents duplicate campaigns from being created. It could be refactored, but as-is this part is necessary.